### DISCUSSION TITLE
argo-workflows/inline-templates/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#9548] [General]
# argo-workflows/inline-templates/

https://argoproj.github.io/argo-workflows/inline-templates/

**Comment by caspereijkens** [comment]
Question about managing Workflows that are generated by WorkflowTemplates.
In ArgoCD, we cannot see the Workflows that are generated by WorkflowTemplates. 
In case of failing Workflows, it would be nice to be able to view/delete them with ArgoCD. How can we do that?

**Reply by OneCricketeer** [reply]
We can see our workflowtemplates from ArgoCD

**Reply by OneCricketeer** [reply]
The workflows themselves exist in the workflow-server, which would be a separate pod/deployment from ArgoCD instance.

**Reply by agilgur5** [reply]
That would be a potential feature request for ArgoCD.

In ArgoCD, you can see Pods of a Deployment, but in that case, the Deployment directly creates the Pods. In this case, a WorkflowTemplate does _not_ create Workflows (and does _not_ have an `ownerReference`).
A Workflow can _reference_ a WorkflowTemplate, similar to how selectors work in Kubernetes, such as with a [PDB](https://kubernetes.io/docs/tasks/run-application/configure-pdb/#specifying-a-poddisruptionbudget).

ArgoCD does have some logic to visualize some relationships, for instance for Services and Ingresses.
But this relationship is a bit different of a case.

I would recommend using the Argo Workflows Server instead, which has a UI to view all Workflows, WorkflowTemplates, and all other Argo Workflows custom resources.

**Comment by gocpplua** [comment]
When I execute hello-world-template-global-arg, the following error occurs:
```
Failed to submit workflow: templates.whalesay.steps[0].hello-world template reference hello-world-template-global-arg.hello-world not found
```

**Reply by gocpplua** [reply]
I have solved, first need to execute `argo template create hello-world-wf-global-arg.yaml -n argo`  to create template, then execute `argo submit hello-world-wf-global-arg.yaml -n argo
` submit workflow

**Comment by katylava** [comment]
In `argo submit --from workflowtemplate/workflow-template-submittable`, I see `workflow-template-submittable` defined in `templates.yaml`, but what does `workflowtemplate` represent? Is that always going to be `workflowtemplate`?

**Reply by joebowbeer** [reply]
`--from kind/name`

Another kind that works here is cronwf

**Comment by dai1975** [comment]
Is there are template languate specifications?
I wonder the engine can treat condition or loop features.

**Reply by OneCricketeer** [reply]
Both are possible. Search "condition" and "loop" in the [examples](https://github.com/argoproj/argo-workflows/tree/master/examples)

**Reply by agilgur5** [reply]
The [Workflow Variables page](https://argoproj.github.io/argo-workflows/variables/) specifies all of the templating features.

**Comment by skight-shen** [comment]
Following the instruction above, I created the workflow template "workflow-template-submittable" from argo ui, then created another workflow template "workflow-template-hello-world-" which refer to "workflow-template-submittable"  in the same namespace. However, it show error "Internal Server Error: template reference workflow-template-hello-world-bwjmb.whalesay-template not found" when submitted a workflow based on template "workflow-template-hello-world-" from UI.

Anybody can help?
